test/auth.test.js

```diff
@@ -0,0 +1,190 @@
+            }); 
+    
+        });
+    
+        describe('signing up with no password', ()=>{
+            it('should return a 500 status code without token', (done)=>{
+                chai.request(server)
+                .post(SIGN_UP_ROUTE)
+                .send({
+                    user:{
+                        username: faker.name.lastName() + Date.now(),
+                        email: faker.internet.email(),
+                    }
+                })
+                .end((err, res)=>{
+                    expect(res.status).to.equal(500);
+                    expect(res.body)
+                       .to.not.have.property('data');
+                    done();
+                });
+            }); 
+    
+        });
+    
+    
+        describe('signing up with all the required fields', ()=>{
+            
+            it('should return a 201 status code has a data property with a token', (done)=>{
+                chai.request(server)
+                .post(SIGN_UP_ROUTE)
+                .send({
+                    user:{
+                        email: faker.internet.email(),
+                        username:faker.name.lastName() + Date.now(),
+                        password: faker.internet.password(),
+                    }
+                })
+                .end((err, res)=>{
+                    expect(res.status).to.equal(201 );
+                    expect(res.body).to.haveOwnProperty('data');
+                    expect(res.body.data).to.haveOwnProperty('token');
+                    done();
+                });
+            }); 
+    
+        });
+    });
+    
+    
+    
+    
+    describe('user on loging in ', ()=>{
+        before((done)=>{
+            chai.request(server)
+            .post(SIGN_UP_ROUTE)
+            .send(tempUser)
+            .end((err, res)=>{
+               done();
+            });
+        });
+       
+        describe('sloging in with no username', ()=>{
+            it('should not return  a data property', (done)=>{
+                chai.request(server)
+                .post(LOGIN_ROUTE)
+                .send({
+                    user:{
+                        password: faker.internet.password()
+                    }
+                })
+                .end((err, res)=>{
+                    expect(res.body)
+                       .to.not.have.property('data');
+                    done();
+                });
+            }); 
+    
+        });
+    
+        describe('loging in with no password', ()=>{
+            it('should not return a data property ', (done)=>{
+                chai.request(server)
+                .post(LOGIN_ROUTE)
+                .send({
+                    user:{
+                        username: faker.name.lastName() + Date.now(),
+                    }
+                })
+                .end((err, res)=>{
+                    expect(res.body)
+                       .to.not.have.property('data');
+                    done();
```

Comment:
For every occurence of
```
.end((err, res)=>{
                    expect(res.body)
                       .to.not.have.property('data');
```
you should also check for the value in `data` object. That way you are sure the value is not the same with what is in

```
                .end((err, res)=>{
                    expect(res.status).to.equal(201 );
                    expect(res.body).to.haveOwnProperty('data');
                    expect(res.body.data).to.haveOwnProperty('token');
                    done();
                })
```